File: BAEKJOON/Javascript/1000/1300/1339.js
/*
Title : 단어 수학
Link : https://www.acmicpc.net/problem/1339
*/

const fs = require("fs");
const filepath =
  process.platform === "linux" ? "/dev/stdin" : __dirname + "/input.txt";
const input = fs.readFileSync(filepath).toString().trim().split("\n");

const N = parseInt(input[0]);
let alps = new Array(26);
for (let i = 0; i < 26; i++) {
  alps[i] = 0;
}
for (let i = 1; i <= N; i++) {
  for (let j = 0; j < input[i].length; j++) {
    alps[input[i].charCodeAt(j) - 65] += 10 ** (input[i].length - j);
  }
}
let tmp = [];
for (let i = 0; i < 26; i++) {
  tmp.push([alps[i], i]);
}
tmp.sort((x, y) => {
  x[0] - y[0];
});
let alpToNum = new Array(26);
for (let i = 0; i < 10; i++) {
  alpToNum[tmp[i][1]] = 9 - i;
}

console.log(alpToNum)

let ans = 0;
for (let i = 1; i <= N; i++) {
  let p = 0;
  for (let j = 0; j <= input[i].length; j++) {
    p *= 10;
    p += alpToNum[input[i].charCodeAt(j)];
  }
  ans += p;
}
console.log(ans);

File: BAEKJOON/Python/1000/1300/1339.py
"""
Title : 단어 수학
Link : https://www.acmicpc.net/problem/1339
"""

from sys import stdin

input = stdin.readline


if __name__ == "__main__":
    N = int(input())
    alps = {chr(65 + x): 0 for x in range(26)}
    numbers = []
    for _ in range(N):
        s = input().strip()
        for i, x in enumerate(s):
            alps[x] += 10 ** (len(s) - i)
        numbers.append(s)
    tmp = sorted([(k, v) for k, v in alps.items() if v], key=lambda x: -x[1])
    alp_to_num = {}
    for i, (k, v) in enumerate(tmp):
        alp_to_num[k] = 9 - i
    ans = 0
    for s in numbers:
        p = 0
        for i, x in enumerate(s):
            p *= 10
            p += alp_to_num[x]
        ans += p
    print(ans)
